server/src/router/federated/server.ts

```diff
@@ -317,13 +316,13 @@ export class Federated extends Server {
 
   /**
    * Appends a request to the logs.
-   * @param taskId The task id for which the request was made
+   * @param task The task id for which the request was made
    * @param nodeId The node id who made the request
```

Comment:
I personally would have preferred naming it `taskId` to be consistent with `nodeId` and to not confuse a task object and a task id when a variable is called `task`. Especially if the function argument you input is `task.id`, then I would be surprised to see the function signature calling it `task` rather than `taskId`.
But I'm nitpicking so I'll let you choose, I saw you renamed it to `task` everywhere so up to you :)

Reply:
I see why it might be confusing. in dynamic typed languages, naming variable with its type is common and help with knowing what to do with the value. but in statically typed languages (such as strict TS), I don't find that it is helping as the type of a variable is known.
in a given function, when using `task` rather than `taskID`, I find it makes the code a bit more readable, as we don't really need to know that it's a full object or an ID, but rather what we'll do with it.
but yeah, that's clearly a matter of taste :)

Reply:
While I agree with Valerian, to be even more picky about this, I would personally name it just id, so that when a variable is called task it's always the actual Task object. This is also useful when looking up the word task over the whole codebase